### ISSUE TITLE
Number Recognition

### ISSUE THREAD

**Issue by CHEVURIVENKATESH2709** [CHEVURIVENKATESH2709/CHEVURIVENKATESH#3]
import numpy as np
import matplotlib.pyplot as plt
from tensorflow.keras.datasets import mnist
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Flatten
from tensorflow.keras.utils import to_categorical
# Load the dataset
(X_train, y_train), (X_test, y_test) = mnist.load_data()

# Preprocess the data
X_train = X_train / 255.0
X_test = X_test / 255.0

# Convert labels to one-hot encoding
y_train = to_categorical(y_train, num_classes=10)
y_test = to_categorical(y_test, num_classes=10)

model = Sequential()
model.add(Flatten(input_shape=(28, 28)))
model.add(Dense(128, activation='relu'))
model.add(Dense(64, activation='relu'))
model.add(Dense(10, activation='softmax'))

model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])

history = model.fit(X_train, y_train, epochs=10, batch_size=32, validation_split=0.1)

loss, accuracy = model.evaluate(X_test, y_test)
print(f"Test Accuracy: {accuracy:.4f}")
# Select a random test image
random_index = np.random.randint(0, len(X_test))
test_image = X_test[random_index]

# Make predictions
predictions = model.predict(np.expand_dims(test_image, axis=0))
predicted_label = np.argmax(predictions)

# Display the image and predicted label
plt.imshow(test_image, cmap='gray')
plt.title(f"Predicted: {predicted_label}")
plt.show()
